### PR TITLE
Switch to using math.comb in Polynomial models. Deprecate modeling.utils.comb

### PR DESCRIPTION
**Pull request by mcara**
### Description
`Polynomial._order` is computed using [`astropy.modeling.utils.comb()`](https://github.com/astropy/astropy/blob/cffe98aaff7a5ffe442934ecdf20b209606db0db/astropy/modeling/utils.py#L218-L235) function which returns floating point numbers, making `_order` - the number of terms in a polynomial and naturally an integer number - a floating point number.

Instead of converting floating point numbers to integers, I propose that `astropy.modeling.utils.comb()` simply performs integer/exact computations.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Do the proposed changes actually accomplish desired goals?
- [ ] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [ ] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [ ] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [ ] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [ ] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [ ] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [ ] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by WilliamJamieson — CHANGES_REQUESTED** [review]
Could you add a test to demonstrate the integer value for `order`.

**Comment by WilliamJamieson** [comment]
> We should be using [`math.comb()`](https://docs.python.org/3/library/math.html#math.comb), which is in the standard library since Python 3.8. Having our own implementation served a purpose for older Python versions.

Agreed.

**Comment by mcara** [comment]
Ha! I wasn't up to speed with `math` developments. Thanks, @eerovaher. I agree with using `math`. I will change/update this PR.

**Comment by WilliamJamieson** [comment]
pre-commit.ci autofix

**Comment by WilliamJamieson** [comment]
@mcara can you squash the pre-commit (e70b275) and build complaint fixes (5fecaf0) commits into one before I merge this?

**Review by pllim — CHANGES_REQUESTED** [review]
I think this needs a change log?

**Comment by pllim** [comment]
Any chance we can squash? 😉

File: docs/changes/modeling/14038.feature.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Deprecated ``astropy.modeling.utils.com()`` function in favor of ``comb()``
```
Comment (pllim) [inline docs/changes/modeling/14038.feature.rst:1]: Typo 😢 
```suggestion
Deprecated ``astropy.modeling.utils.comb()`` function in favor of ``comb()``
```

Comment (mcara) [reply docs/changes/modeling/14038.feature.rst:1]: 😭

**Review by pllim — APPROVED** [review]
Thanks!

**Comment by mcara** [comment]
Phew! 💦

**Pull request merged**